lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/include/Config/Serializable.hpp

```diff
@@ -0,0 +1,275 @@
+     * serializable and deserializable from JSON.
+     */
+    struct Serializable
+    {
+    protected:
+        ~Serializable() = default;
+
+        /**
+         * \brief Function that is called when an element inside the json object
+         * passed to from_json is being parsed.
+         * \param name The name of the JSON element.
+         * \param value The value of the JSON element.
+         */
+        virtual void parse_json_element(const std::string& name, json_value_s& value) = 0;
+
+        /**
+         * \brief Parses a JSON array into an std::vector of a specified type T.
+         * \tparam T The type for to parse a vector of from the given json value.
+         * \param array_value The JSON value containing the JSON array.
+         * \return An std::vector<T>
+         */
+        template<typename T>
+        static std::enable_if_t<
+            std::is_base_of_v<Serializable, T> || 
+            std::is_same_v<std::string, T>
+        , std::vector<T>> parse_json_array(json_value_s& array_value)
+        {
+            std::vector<T> elements;
+
+            const auto array = json_value_as_array(&array_value);
+
+            if (array == nullptr)
+            {
+                return elements;
+            }
+
+            auto array_iterator = array->start;
+
+            while(array_iterator)
+            {
+                auto element = T();
+
+                if constexpr(std::is_same_v<T, std::string>)
+                {
+                    const auto temp_string = json_value_as_string(array_iterator->value);
+                    element = (temp_string == nullptr) ? "" : temp_string->string;
+                }
+                else if constexpr(std::is_base_of_v<Serializable, T>)
+                {
+                    element.from_json(*(array_iterator->value));
+                }
+                else
+                {
+                    throw;
+                }
+
+                elements.push_back(element);
+
+                array_iterator = array_iterator->next;
+            }
+
+            return elements;
+        }
+
+        /**
+         * \brief Parses a number from json into a specific type. 
+         * \tparam T The type to parse the number into. Supports int, double,
+         * long, float, uint64_t, and uint32_t
+         * \param json_value The json value that contains a number.
+         * \return The result of parsing the json value into the specified number type.
+         */
+        template <typename T, typename = std::enable_if_t<
+                      std::is_same_v<T, int> || 
+                      std::is_same_v<T, double> || 
+                      std::is_same_v<T, long> ||
+                      std::is_same_v<T, float> ||
+                      std::is_same_v<T, uint64_t> ||
+                      std::is_same_v<T, uint32_t>
+                  >>
+        static T parse_number(json_value_s& json_value)
+        {
+            T number_value;
```

Comment:
The reason it does not need to be assigned a value is because the template parameter constraints require it to be a value-type - so it has a default value of `0`. That said, explicitly initializing the value will prevent potential issues if the function is extended or modified in the future - so I changed it to initialize to 0.

Reply:
It's my understanding that the default value for builtin types on the stack (i.e. non-static objects) is that they are not initialized in C / C++ so it's value would be what happens to be on that part of the stack; at least as of C++11 / C11.

Reply:
This is not true for C++17 in the types that the template is restricted to - but regardless - and for clarity I changed the implementation to initialize the value to `0`.

Reply:
The default value for int, float, and double, I believe is still uninitialized data.